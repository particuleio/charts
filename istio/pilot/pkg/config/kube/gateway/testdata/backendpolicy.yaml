apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: default
spec:
  gatewayClassName: istio
  listeners:
  - port: 34000
    protocol: TCP
    routes:
      selector:
        matchLabels:
          selected: "yes"
      kind: TCPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: TCPRoute
metadata:
  name: tcp
  namespace: default
  labels:
    selected: "yes"
spec:
  rules:
  - forwardTo:
    - serviceName: httpbin1
      port: 7070
    - serviceName: httpbin2
      port: 9090
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: BackendPolicy
metadata:
  name: httpbin1
  namespace: default
spec:
  backendRefs:
    - name: httpbin1
      group: core
  tls:
    certificateAuthorityRef:
      name: ca-cert
      group: core
      kind: Secret
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: BackendPolicy
metadata:
  name: httpbin2
  namespace: default
spec:
  backendRefs:
    - name: httpbin
      group: core
      port: 80
  tls:
    certificateAuthorityRef:
      name: ca-cert
      group: core
      kind: Secret
