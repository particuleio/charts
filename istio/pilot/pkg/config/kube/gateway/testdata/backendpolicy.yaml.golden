apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*'
    port:
      name: tcp-34000-gateway-gateway-default
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: tcp-tcp-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - default/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: httpbin1.default.svc.domain.suffix
        port:
          number: 7070
      weight: 50
    - destination:
        host: httpbin2.default.svc.domain.suffix
        port:
          number: 9090
      weight: 50
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  creationTimestamp: null
  name: httpbin1-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  host: httpbin1.default.svc.domain.suffix
  trafficPolicy:
    tls:
      credentialName: ca-cert
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  creationTimestamp: null
  name: httpbin2-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  host: httpbin.default.svc.domain.suffix
  trafficPolicy:
    portLevelSettings:
    - port:
        number: 80
      tls:
        credentialName: ca-cert
        mode: SIMPLE
---
